### ISSUE TITLE
docker parity keeps crashing 

### ISSUE THREAD

**Issue by grigored** [paritytech/parity-ethereum#9810]
_Before filing a new issue, please **provide the following information**._

- **Parity Ethereum version**: 0.0.0
- **Operating system**: Amazon Linux
- **Installation**: docker
- **Fully synchronized**: no
- **Network**: ethereum
- **Restarted**: yes

_Your issue description goes here below. Try to include **actual** vs. **expected behavior** and **steps to reproduce** the issue._

I'm starting parity using the following simple docker-compose configuration.
```
  ethnode:
    image: parity/parity:stable
    volumes:
    - /home/ec2-user/nodeData/ethData:/root/.local/share/io.parity.ethereum
    command: --unsafe-expose --ui-no-validation --chain=foundation --mode=active --no-ws --no-ipc --pruning=fast --pruning-memory=2000 --min-peers=25 --max-peers=50
``` 

Yet every few minutes the container is killed and has to restart syncing. I only see this in logs:
```
2018-10-25 12:01:32 UTC Syncing #6578024 0x93b8…3142     1.22 blk/s   99.4 tx/s    8 Mgas/s      0+  562 Qed  #6578587    5/25 peers      5 MiB chain    2 GiB db   47 MiB queue   13 MiB sync  RPC:  0 conn,    0 req/s,    0 µs
./entrypoint.sh: line 2:     6 Killed                  parity $@
```

How do I get parity to work?

**Comment by joshua-mir** [comment]
Could you perhaps set a `--log-file` to help figure out what is going on?

**Comment by grigored** [comment]
@joshua-mir thanks for the fast response. I set a log file, but that doesn't show more data (below). I upgraded to v2.1.3, and that didn't help either. I'm running this on a t2.medium aws instance (4g memory, 400GB ssd, 1200 iops). Could that be the problem? I'm retrying on a bigger t2.xlarge instance (16gb, will let you know how it goes). Thanks!
```
2018-10-27 14:41:32 UTC Starting Parity-Ethereum/v2.1.3-beta-18ddd7c-20181015/x86_64-linux-gnu/rustc1.29.0
2018-10-27 14:41:32 UTC Keys path /home/parity/.local/share/io.parity.ethereum/keys/ethereum
2018-10-27 14:41:32 UTC DB path /home/parity/.local/share/io.parity.ethereum/chains/ethereum/db/906a34e69aec8c0d
2018-10-27 14:41:32 UTC State DB configuration: fast
2018-10-27 14:41:32 UTC Operating mode: active
2018-10-27 14:41:32 UTC Configured for Foundation using Ethash engine
2018-10-27 14:41:58 UTC Updated conversion rate to Ξ1 = US$202.22 (23548140 wei/gas)
2018-10-27 14:42:02 UTC Public node URL: enode://382297e588711436edf34e3b2d2a77012ef188e5a3d29425b53821cc52e96291da32c79cae3308136d58ee07fd5f14134e6eff026d7d1ce88b3ae58cb9f6a023@172.31.5.87:30303
2018-10-27 14:42:14 UTC Starting Parity-Ethereum/v2.1.3-beta-18ddd7c-20181015/x86_64-linux-gnu/rustc1.29.0
2018-10-27 14:42:14 UTC Keys path /home/parity/.local/share/io.parity.ethereum/keys/ethereum
2018-10-27 14:42:14 UTC DB path /home/parity/.local/share/io.parity.ethereum/chains/ethereum/db/906a34e69aec8c0d
2018-10-27 14:42:14 UTC State DB configuration: fast
2018-10-27 14:42:14 UTC Operating mode: active
2018-10-27 14:42:15 UTC Configured for Foundation using Ethash engine
2018-10-27 14:42:40 UTC Updated conversion rate to Ξ1 = US$202.22 (23548140 wei/gas)
```

**Comment by grigored** [comment]
ran with ` -l sync=trace`, and here are the logs right before it crashes, not sure if it helps:
```
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  New header: d85a929b209153c42ee9d167e28798cf00773b8fc1c38f44670dc2b7a9643265
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  New header: 0e749c181dbea2250289eefd5476cd716405c672fd0eaa066c736697661671ca
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  New header: 254909b356a238a1cbd0a14aa62618783bf1f0a41d32862283a8118da5ebb743
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  New header: dda36b7b4db194ea1687766a3885f28f9d768e591c5847f791c608abb60152f1
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  Completed subchain 0x55588b7bb2d34ec7965f59f59ba055125f2766012face82f5eb456bcc5909d88
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  NewBlocks: Inserted 126 headers
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  Drained 0 blocks, new head :Some(0x5bc70a565c2da975e203d25cf350b481461380b53a47391f90ba080a990b542d)
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  NewBlocks: Imported 0 of 0
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  Considering peer 0, force=false, td=Some(7471027032202655789916), our td=7467733111775788829262, latest=0x3956…7c7c, have_latest=false, state=Blocks
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  Syncing with peer 0, force=false, td=Some(7471027032202655789916), our td=7467733111775788829262, state=Blocks
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  0 <- GetBlockBodies: 32 entries starting from Some(0x55295840250052068898b344e556e5b8241bf7b7f8b114570405428ad29a3159), set = NewBlocks
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  Syncing with peers: 3 active, 4 confirmed, 4 total
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  Skipping busy peer 0
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  Skipping busy peer 1
2018-10-27 15:06:55 UTC IO Worker #3 TRACE sync  Skipping busy peer 2
2018-10-27 15:06:55 UTC IO Worker #1 DEBUG sync  2 -> Dispatching packet: 4
2018-10-27 15:06:55 UTC IO Worker #1 TRACE sync  2 -> BlockHeaders (1 entries), state = Blocks, set = NewBlocks
2018-10-27 15:06:55 UTC IO Worker #1 TRACE sync  NewBlocks: Skipping existing block header 6593778 (0x3956c2de70b3a7e1d0c0ce77f0cfeadd3f639850de46c0ff78a84840e71a7c7c)
2018-10-27 15:06:55 UTC IO Worker #1 TRACE sync  NewBlocks: No useful headers (2 this round), expected hash 0x3956c2de70b3a7e1d0c0ce77f0cfeadd3f639850de46c0ff78a84840e71a7c7c
2018-10-27 15:06:55 UTC IO Worker #1 TRACE sync  Deactivating peer 2
2018-10-27 15:06:55 UTC IO Worker #1 TRACE sync  Syncing with peers: 2 active, 4 confirmed, 4 total
2018-10-27 15:06:55 UTC IO Worker #1 TRACE sync  Skipping busy peer 1
2018-10-27 15:06:55 UTC IO Worker #1 TRACE sync  Skipping busy peer 0
./entrypoint.sh: line 2:     6 Killed                  parity $@
```

**Comment by joshua-mir** [comment]
@grigored it might be more helpful to run `-l shutdown=trace` or `-l debug`.

**Comment by grigored** [comment]
Actually I had four problems before:
a) the node suddenly crashed, and then on each restart would crash again after a few seconds.
b) crashed and started syncing snapshots and got stuck after snapshot 1700/2200 
c) it never loaded snapshots from disk, always from network.
d) only connected to a maximum of 4-5 peers. This was fixed after switching to 

It seems to work now. 

I switched to an ec2 t2.xlarge instance instead of t2.medium, opened port 30303 which apparently was closed before, and run parity:v2.1.3 instead of parity:stable.  
c) was fixed after using parity:v2.1.3 instead of parity:stable. 
d) was fixed after opening port 30303. Don't understand how come the node was connecting to some peers even with 30303 closed, but not more than 5.

Not sure what fixed a) and b)
